pkg/csi/service/wcp/controller.go

```diff
@@ -1509,7 +1550,24 @@ func (c *controller) ControllerExpandVolume(ctx context.Context, req *csi.Contro
 		// If thr reqeust failed due to object not found, "csi.fault.NotFound" will be return.
 		// For all other cases, the faultType will be set to "csi.fault.Internal" for now.
 		// Later we may need to define different csi faults.
+		// Check if the volume contains CNS snapshots only for block volumes.
```

Comment:
Do we need to determine volume type here like what you did in DeleteVolume code?

Reply:
Yes, good catch, fixed it.